lib/db/mysql.js

```diff
@@ -1082,5 +1084,46 @@ module.exports = function (log, error) {
       )
   }
 
+  function ipHmac(key, uid, addr) {
+    if (ip.isV4Format(addr)) {
+      addr = '::' + addr
+    }
+    addr = ip.toBuffer(addr)
+
+    var hmac = crypto.createHmac('sha256', key)
+    hmac.update(uid)
+    hmac.update(addr)
+
+    return hmac.digest()
+  }
+
+  var SECURITY_EVENT_NAMES = {
+    'account.create': 1,
+    'account.login': 2,
+    'account.reset': 3
+  }
+
+  var CREATE_SECURITY_EVENT = 'CALL createSecurityEvent_1(?, ?, ?, ?, ?)'
+  MySql.prototype.createSecurityEvent = function (data) {
+    var uid = data.uid
+    var tokenId = data.tokenId
+    var nameId = SECURITY_EVENT_NAMES[data.name]
+    var ipAddr = ipHmac(this.ipHmacKey, uid, data.ipAddr)
+    return this.write(CREATE_SECURITY_EVENT, [uid, tokenId, nameId, ipAddr, Date.now()])
+  }
+
+  var FETCH_SECURITY_EVENTS = 'CALL fetchSecurityEvents_1(?, ?)'
+  MySql.prototype.securityEvents = function (where) {
+    var uid = where.uid
+
+    var ipAddr = ipHmac(this.ipHmacKey, uid, where.ipAddr)
```

Comment:
As part of geo-profiling, it will be important to look at all security events for a `uid`. Is it possible to extend this query to support returning all security events? Possibly with a timestamp filter?